.github/workflows/build-and-push-docker-hub.yaml

```diff
@@ -1,52 +1,460 @@
-name: Build and Push Docker Image to Docker Hub
+name: Build, Attest, Sign and publish Docker Image
 
 on:
-  push:
-    tags:
-      - "v*.*.*"
+  workflow_call:
+    outputs:
+      image-ref:
+        description: "Full image reference that was pushed"
+        value: ${{ jobs.build-and-push.outputs.image-ref }}
+      image-digest:
+        description: "Content digest of the pushed image"
+        value: ${{ jobs.build-and-push.outputs.image-digest }}
+      oci-artifact-name:
+        description: "Name of the exported runtime OCI artifact"
+        value: ${{ jobs.build-and-push.outputs.oci-artifact-name }}
+      platforms:
+        description: "Resolved platform list used for the build"
+        value: ${{ jobs.build-and-push.outputs.platforms }}
+    inputs:
+      image_name:
+        description: "Docker image name (e.g. user/repo)"
+        required: true
+        type: string
+      image_tags:
+        description: "Comma-separated list of tags"
+        required: true
+        type: string
+      dockerfile:
+        description: "Path to the Dockerfile"
+        required: true
+        type: string
```

Comment:
Why do we need this argument? Shouldn't we always be using the Dockerfile from this repository?
```suggestion
```